src/components/AddressResolver/EndpointEntry/EndpointEntry.tsx

```diff
@@ -147,133 +147,137 @@ export const EndpointEntry: FunctionComponent<EndpointEntryProps> = ({
+                onChange={onEndpointApiChanged}
+                errorMessage={inputErrorMessageEndpoint}
+              />
+            ) : (
+              <>{api}</>
+            )}
+          </div>
         </div>
 
-        <div className="flex flex-col px-3 pt-3 lg:flex-row lg:w-2/12 lg:px-0 lg:pt-0">
-          <div className="inline-block text-cloud-900 text-xl font-bold lg:hidden">API Header</div>
-          {isEditable ? (
-            <Input
-              className="w-full lg:w-10/12"
-              placeholder="API Header"
-              value={endpointApiHeader}
-              onChange={onEndpointApiHeaderChanged}
-              errorMessage={inputErrorMessageApiHeader}
-            />
-          ) : (
-            <>{apiHeader}</>
-          )}
+        <div className="w-full lg:w-2/12 px-2">
+          <div className="my-2 lg:my-0">
+            <h4 className="text-cloud-900 text-xl font-bold lg:hidden">API Header</h4>
+            {isEditable ? (
+              <Input
+                className="w-full"
+                placeholder="API Header"
+                value={endpointApiHeader}
+                onChange={onEndpointApiHeaderChanged}
+                errorMessage={inputErrorMessageApiHeader}
+              />
+            ) : (
+              <>{apiHeader}</>
+            )}
+          </div>
         </div>
 
-        <div className="flex flex-col p-3 lg:flex-row lg:w-2/12 lg:p-0">
-          <div className="inline-block text-cloud-900 text-xl font-bold lg:hidden">API Key</div>
-          {isEditable ? (
-            <Input
-              className="w-full lg:w-10/12"
-              placeholder="API Key"
-              value={endpointApiKey}
-              onChange={onEndpointApiKeyChanged}
-              errorMessage={inputErrorMessageApiKey}
-            />
-          ) : (
-            <>{apiKey}</>
-          )}
+        <div className="w-full lg:w-2/12 px-2">
+          <div className="my-2 lg:my-0">
+            <h4 className="text-cloud-900 text-xl font-bold lg:hidden">API Key</h4>
+            {isEditable ? (
+              <Input
+                className="w-full"
+                placeholder="API Key"
+                value={endpointApiKey}
+                onChange={onEndpointApiKeyChanged}
+                errorMessage={inputErrorMessageApiKey}
+              />
+            ) : (
+              <>{apiKey}</>
+            )}
+          </div>
         </div>
 
-        {!isEditable && (
-          <div className="flex absolute text-cerulean-200 ml-60 -mt-7 lg:static lg:w-1/12 lg:m-0">
-            <Trash2 className="cursor-pointer" onClick={removeEndpoint} data-testid="trash2-icon" />
-            <Edit
-              className="ml-3 cursor-pointer"
-              onClick={() => {
-                setEditable(true);
-              }}
-              data-testid="edit-icon"
-            />
-          </div>
-        )}
+        <div className="w-auto ml-auto px-2 absolute top-0 right-0 lg:relative">
```

Comment:
maybe can reduce the width of the div wrapping trash/edit icons then can give Endpoint more space since urls are usually longer in length.

Reply:
yeah agree on urls tend to be longer, but the space is tight and limited atm. a bit tough to allocate. will leave the icons as it is (already very small) any smaller might be hard to click